### PR TITLE
Raise issue when CI on `main` fails

### LINKED ISSUE #43: Raise issue when CI on `main` fails
If CI fails for some reason on a push to `main`, it'd be good for us to have that visible (today there's just a red X that most folks won't notice). Dask's `upstream` CI build opens an issue when there's a failure on `main` and leaves a comment linking to the corresponding CI build. We can reuse that GHA workflow here.

### PR DESCRIPTION
**Pull request by ncclementi**
This probably has problems, and I'm not quite sure how to test it works. But dropping something here to get it started

- [x] Closes #43

### CODE REVIEW

**Comment by ncclementi** [comment]
I'm trying to make this Fail on purpose and also making a condition to report on PR, to see if it works. But It seems the report workflow didn't run and it was skipped. Not sure what I'm doing wrong yet, any ideas are welcome!

File: .github/workflows/conda.yml
Comment (jrbourbeau) [inline .github/workflows/conda.yml:152]: xref https://github.com/dask/dask/blob/bfc76afdcdc43c575a10ffefda94aaba424fe347/.github/workflows/upstream.yml#L75

```suggestion
      && needs.conda.result == 'failure'
```

**Comment by jrbourbeau** [comment]
✨ https://github.com/coiled/coiled-runtime/issues/59 ✨

File: .github/workflows/conda.yml
Comment (jrbourbeau) [inline .github/workflows/conda.yml:164]: Nit: Even though this has less content, it somehow seems more clear to me. What do you think? 

```suggestion
            const title = "⚠️ CI failed ⚠️"
```

**Comment by ncclementi** [comment]
Reverted the Intentional Fail. 
Once this is green I think it is ready to merge, unless we need to add anything.

**Review by jrbourbeau — APPROVED** [review]
This looks great -- thanks @ncclementi!

**Pull request merged**
